sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/PodInfoBuilder.java

```diff
@@ -322,6 +322,15 @@ public void setExecutorVolume(VolumeSpec volumeSpec) {
 
     taskInfoBuilder.setContainer(getContainerInfo(podInstance.getPod(), true, true));
 
+    if (taskSpec.getSharedMemory().isPresent()) {
```

Comment:
Question about improper/invalid ServiceSpec's

If we have `podSpec.getSharedMemory().isPresent() == false` and we set 
`taskInfoBuilder.getContainerBuilder().getLinuxInfoBuilder().setIpcMode(taskSpec.getSharedMemory().get()).build()`
due to bad ServiceSpec, will Mesos reject this?

I think it might be worthwhile to extend and add validation for SHM in [DefaultConfigValidators](https://github.com/mesosphere/dcos-commons/blob/0f532c3c76f7826c028248f9985731423ace207d/sdk/scheduler/src/main/java/com/mesosphere/sdk/config/validate/DefaultConfigValidators.java#L17) to filter out these types of cases.

Reply:
added a new config validator for taskSpec validation